rust/core-lib/src/editor.rs

```diff
@@ -622,8 +622,9 @@ impl Editor {
                 if let Some(end) = self.text.next_grapheme_offset(middle) {
                     if start >= last {
                         let interval = Interval::new_closed_open(start, end);
-                        let swapped = self.text.slice_to_string(middle..end) +
-                                      &self.text.slice_to_string(start..middle);
+                        let swapped: String = format!("{}{}",
```

Comment:
A perhaps slightly cleaner way to write this would be,
```rust
let before =  self.text.slice_to_string(start..middle);
let after = self.text.slice_to_string(middle..end);
let swapped: String = [after, before].concat();
```

(or some variant on that; but the basic pattern of `concat`ing an array.)